game/scripts/vscripts/TutorialGraph/Steps.ts

```diff
@@ -1,15 +1,27 @@
-import { getCameraDummy, findAllPlayersID, getPlayerHero, setGoalsUI, setUnitVisibilityThroughFogOfWar, createPathParticle, getOrError, HighlightProps, highlight } from "../util"
+import { getCameraDummy, findAllPlayersID, getPlayerHero, setGoalsUI, setUnitVisibilityThroughFogOfWar, createPathParticle, getOrError, HighlightProps, highlight, unitIsValidAndAlive } from "../util"
 import * as dg from "../Dialog"
 import * as tg from "./Core"
 import { getSoundDuration } from "../Sounds"
 
-const isHeroNearby = (location: Vector, radius: number) => FindUnitsInRadius(
-    DotaTeam.BADGUYS, location, undefined, radius,
-    UnitTargetTeam.BOTH,
-    UnitTargetType.HERO,
-    UnitTargetFlags.INVULNERABLE + UnitTargetFlags.OUT_OF_WORLD + UnitTargetFlags.MAGIC_IMMUNE_ENEMIES,
-    0, false
-).length > 0
+const isPlayerHeroNearby = (location: Vector, radius: number) => {
+    const unitsWithinRadius = FindUnitsInRadius(
+        DotaTeam.BADGUYS, location, undefined, radius,
+        UnitTargetTeam.BOTH,
+        UnitTargetType.HERO,
+        UnitTargetFlags.INVULNERABLE + UnitTargetFlags.OUT_OF_WORLD + UnitTargetFlags.MAGIC_IMMUNE_ENEMIES,
+        0, false
+    )
+    
+    let foundHeroUnit = false
+    const playerHeroName = getOrError(getPlayerHero()).GetName()
+
+    for (const unitWithinRadius of unitsWithinRadius) {
+        if (unitWithinRadius.GetName() === playerHeroName)
+            foundHeroUnit = true
+    }
+
+    return foundHeroUnit
```

Comment:
Can just do something like `getOrError(getPlayerHero()).GetAbsOrigin().__sub(location).Length2D() < radius` instead of all this.

Reply:
Nice that's much better, added it